src/geo/map/cartodb-layer.js

```diff
@@ -147,6 +147,16 @@ var CartoDBLayer = LayerModelBase.extend({
   remove: function () {
     this.getSource().unmarkAsSourceOf(this);
     LayerModelBase.prototype.remove.apply(this, arguments);
+  },
+
+  $getTableName: function () {
+    if (this.attributes.source.attributes.options) {
```

Comment:
Two things:

- Instead of accessing `this.attributes` use the `get` method (general tip for Backbone models). In this case, you can use the existing `getSource()`.
- This is an internal class and we are not following the `$` convention that we use in the public API.